core/src/main/java/org/apache/gravitino/storage/relational/JDBCBackend.java

```diff
@@ -210,32 +231,45 @@ public <E extends Entity & HasIdentifier> E get(
-  public boolean delete(NameIdentifier ident, Entity.EntityType entityType, boolean cascade)
-      throws IOException {
-    switch (entityType) {
-      case METALAKE:
-        return MetalakeMetaService.getInstance().deleteMetalake(ident, cascade);
-      case CATALOG:
-        return CatalogMetaService.getInstance().deleteCatalog(ident, cascade);
-      case SCHEMA:
-        return SchemaMetaService.getInstance().deleteSchema(ident, cascade);
-      case TABLE:
-        return TableMetaService.getInstance().deleteTable(ident);
-      case FILESET:
-        return FilesetMetaService.getInstance().deleteFileset(ident);
-      case TOPIC:
-        return TopicMetaService.getInstance().deleteTopic(ident);
-      case USER:
-        return UserMetaService.getInstance().deleteUser(ident);
-      case GROUP:
-        return GroupMetaService.getInstance().deleteGroup(ident);
-      case ROLE:
-        return RoleMetaService.getInstance().deleteRole(ident);
-      case TAG:
-        return TagMetaService.getInstance().deleteTag(ident);
-      default:
-        throw new UnsupportedEntityTypeException(
-            "Unsupported entity type: %s for delete operation", entityType);
+  public boolean delete(NameIdentifier ident, Entity.EntityType entityType, boolean cascade) {
+    // Invalidate the cache first
+    EntityIdentifier entityIdentifier = EntityIdentifier.of(ident, entityType);
+    NameIdMappingService.getInstance().invalidate(entityIdentifier);
+    if (cascade) {
+      // Remove all the children entities in the cache;
+      NameIdMappingService.getInstance().invalidateWithPrefix(entityIdentifier);
+    }
+
+    try {
+      switch (entityType) {
+        case METALAKE:
+          return MetalakeMetaService.getInstance().deleteMetalake(ident, cascade);
+        case CATALOG:
+          return CatalogMetaService.getInstance().deleteCatalog(ident, cascade);
+        case SCHEMA:
+          return SchemaMetaService.getInstance().deleteSchema(ident, cascade);
+        case TABLE:
+          return TableMetaService.getInstance().deleteTable(ident);
+        case FILESET:
+          return FilesetMetaService.getInstance().deleteFileset(ident);
+        case TOPIC:
+          return TopicMetaService.getInstance().deleteTopic(ident);
+        case USER:
+          return UserMetaService.getInstance().deleteUser(ident);
+        case GROUP:
+          return GroupMetaService.getInstance().deleteGroup(ident);
+        case ROLE:
+          return RoleMetaService.getInstance().deleteRole(ident);
+        case TAG:
+          return TagMetaService.getInstance().deleteTag(ident);
+        default:
+          throw new UnsupportedEntityTypeException(
+              "Unsupported entity type: %s for delete operation", entityType);
+      }
+    } finally {
+      // Remove the entity from the cache again because we may add the cache during the deletion
+      // process
+      NameIdMappingService.getInstance().invalidateWithPrefix(entityIdentifier);
```

Comment:
Maybe we should follow the pattern.
We write the slow storage first, then write the fast storage.

Reply:
Could you provide more details for the sentence `We write the slow storage first, then write the fast storage.`?